pandas/core/index.py

```diff
@@ -111,9 +112,31 @@ def __new__(cls, data=None, dtype=None, copy=False, name=None, fastpath=False,
         if fastpath:
             return cls._simple_new(data, name)
 
+        # RangeIndex pass-through
+        # Index(start, stop, ...) --> RangeIndex(start, stop, ...)
+        if isinstance(data, int):
+            if dtype is None and copy == False:
+                copy = None
+                range_constructor = True
+            elif isinstance(dtype, int):
+                range_constructor = True
+                if copy == False:
+                    copy = None
+                elif isinstance(copy, int):
+                    range_constructor = True
+                else:
```

Comment:
ok, see my other comment. if data is a scalar, then its an error (and btw use ``lib.isscalar``)